ksql-serde/src/main/java/io/confluent/ksql/serde/Format.java

```diff
@@ -86,18 +46,37 @@ public KsqlSerdeFactory getSerdeFactory(final FormatInfo info) {
    *
    * @return whether or not this format supports unwrapping
    */
-  public boolean supportsUnwrapping() {
-    return supportsUnwrapping;
+  default boolean supportsUnwrapping() {
+    return false;
```

Comment:
Having written this code I now know that this should be the other way around, i.e. it should be `supportsWrapping`, which effectively means 'it supports wrapping multiple columns into fields of some anonymous outer object', e.g. like a JSON object or a Protobuf message.

Unfortunately, I've never found the time to flip this.  But it would clean up some nasties in the code.  

If we're making this a public API then I'd flip this first.